examples/five_by_five.py

```diff
@@ -0,0 +1,273 @@
+                )
+                if moves > self.MIN_MOVES
+                else " Well done! That's the minimum number of moves to solve the puzzle!"
+            )
+        )
+        self.add_class("visible")
+
+    def hide(self) -> None:
+        """Hide the winner message."""
+        self.remove_class("visible")
+
+
+class GameHeader(Widget):
+    """Header for the game.
+
+    Comprises of the title (``#app-title``), the number of moves ``#moves``
+    and the count of how many cells are turned on (``#progress``).
+    """
+
+    #: Keep track of how many moves the player has made.
+    moves = reactive(0)
+
+    #: Keep track of how many cells are turned on.
+    on = reactive(0)
+
+    def compose(self) -> ComposeResult:
+        """Compose the game header."""
+        yield Horizontal(
+            Static(self.app.title, id="app-title"),
+            Static(id="moves"),
+            Static(id="progress"),
+        )
+
+    def watch_moves(self, moves: int):
+        """Watch the moves reactive and update when it changes."""
+        self.query_one("#moves", Static).update(f"Moves: {moves}")
+
+    def watch_on(self, on: int):
+        """Watch the on-count reactive and update when it changes."""
+        self.query_one("#progress", Static).update(f"On: {on}")
+
+
+class GameCell(Button):
+    """Individual playable cell in the game."""
+
+    @staticmethod
+    def at(row: int, col: int) -> str:
+        return f"cell-{row}-{col}"
+
+    def __init__(self, row: int, col: int) -> None:
+        """Initialise the game cell."""
+        super().__init__("", id=self.at(row, col))
+        self.row = row
+        self.col = col
+
+
+class GameGrid(Widget):
+    """The main playable grid of game cells."""
+
+    def compose(self) -> ComposeResult:
+        """Compose the game grid."""
+        for row in range(Game.SIZE):
+            for col in range(Game.SIZE):
+                yield GameCell(row, col)
+
+
+class Game(Screen):
+    """Main 5x5 game grid screen."""
+
+    #: The size of the game grid. Clue's in the name really.
+    SIZE = 5
+
+    #: The bindings for the main game grid.
+    BINDINGS = [
+        Binding("n", "new_game", "New Game"),
+        Binding("question_mark", "push_screen('help')", "Help", key_display="?"),
+        Binding("q", "quit", "Quit"),
+        Binding("up,w,k", "navigate(-1,0)", "Move Up", False),
+        Binding("down,s,j", "navigate(1,0)", "Move Down", False),
+        Binding("left,a,h", "navigate(0,-1)", "Move Left", False),
+        Binding("right,d,l", "navigate(0,1)", "Move Right", False),
+        Binding("space", "move", "Toggle", False),
+    ]
+
+    @property
+    def on_cells(self) -> DOMQuery[GameCell]:
+        """The collection of cells that are currently turned on.
+
+        :type: DOMQuery[GameCell]
+        """
+        return cast(DOMQuery[GameCell], self.query("GameCell.on"))
+
+    @property
+    def on_count(self) -> int:
```

Comment:
The naming of this conflicts with our event handlers. Can we rename to prevent any confusion?